toolkits/slack/arcade_slack/tools/chat.py

```diff
@@ -72,43 +92,508 @@ def send_dm_to_user(
+            if channel["name"].lower() == channel_name.lower():
+                channel_id = channel["id"]
+                break
+
+        if not channel_id:
+            raise RetryableToolError(
+                "Channel not found",
+                developer_message=f"Channel with name '{channel_name}' not found.",
+                additional_prompt_content=format_conversations_as_csv(channels_response),
+                retry_after_ms=500,  # Play nice with Slack API rate limits
+            )
+
+        # Step 2: Send the message to the channel
+        response = await slackClient.chat_postMessage(channel=channel_id, text=message)
+
+    except SlackApiError as e:
+        error_message = e.response["error"] if "error" in e.response else str(e)
+        raise ToolExecutionError(
+            "Error sending message",
+            developer_message=f"Slack API Error: {error_message}",
+        )
+    else:
+        return {"response": response.data}
+
+
+@tool(
+    requires_auth=Slack(
+        scopes=["channels:read", "groups:read", "im:read", "mpim:read"],
     )
+)
+async def get_members_from_conversation_by_id(
+    context: ToolContext,
+    conversation_id: Annotated[str, "The ID of the conversation to get members for"],
+    limit: Annotated[
+        Optional[int], "The maximum number of members to return."
+    ] = MAX_PAGINATION_SIZE_LIMIT,
+    next_cursor: Annotated[Optional[str], "The cursor to use for pagination."] = None,
+) -> Annotated[dict, "Information about each member in the conversation"]:
+    """Get the members of a conversation in Slack by the conversation's ID."""
+    slackClient = AsyncWebClient(token=context.authorization.token)
 
-    # Step 1: Retrieve the list of channels
-    channels_response = slackClient.conversations_list()
-    channel_id = None
-    for channel in channels_response["channels"]:
-        if channel["name"].lower() == channel_name.lower():
-            channel_id = channel["id"]
-            break
+    try:
+        member_ids, next_cursor = await async_paginate(
+            slackClient.conversations_members,
+            "members",
+            limit=limit,
+            next_cursor=next_cursor,
+            channel=conversation_id,
+        )
+    except SlackApiError as e:
+        if e.response["error"] == "channel_not_found":
+            conversations = await list_conversations_metadata(context)
+            available_conversations = ", ".join(
+                f"{conversation['id']} ({conversation['name']})"
+                for conversation in conversations["conversations"]
+            )
 
-    if not channel_id:
-        raise RetryableToolError(
-            "Channel not found",
-            developer_message=f"Channel with name '{channel_name}' not found.",
-            additional_prompt_content=format_channels(channels_response),
-            retry_after_ms=500,  # Play nice with Slack API rate limits
+            raise RetryableToolError(
```

Comment:
I'm curious how beneficial it is to use a RetryableToolError for this tool. For names, it can be helpful to use because it's obvious that "Johhn" should be retried as "John", but with IDs, I'm not sure if this applies.